ui/src/app/settings/components/cluster-details/cluster-details.tsx

```diff
@@ -25,7 +25,7 @@ export const ClusterDetails = (props: RouteComponentProps<{server: string}>) =>
     const loaderRef = React.useRef<DataLoader>();
     const [updating, setUpdating] = React.useState(false);
     return (
-        <DataLoader ref={loaderRef} input={server} load={(url: string) => timer(0, 1000).pipe(mergeMap(() => from(services.clusters.get(url, ''))))}>
+        <DataLoader ref={loaderRef} input={server} load={(name: string) => timer(0, 5000).pipe(mergeMap(() => from(services.clusters.getByName(name))))}>
```

Comment:
Indeed, because the Cluster API URL is not unique anymore, so we use the Cluster Name.
We could eventually make it unique without breaking compatibility by relying on Kubernetes name+namespace uniqueness by following what Traefik does: encoding the name into something like `resource-name@namespace`